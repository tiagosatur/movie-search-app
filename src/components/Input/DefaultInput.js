import React from 'react';
import styled, { css } from 'styled-components';
import { colorPallete, rem } from '../../style';

const DefaultInput = props => {
  const {
    classes,
    placeholder,
    id,
    name,
    value,
    disabled,
    type,
    handleKeyUp,
    handleChange,
    submit,
  } = props;

  return (
    <StyledDefaultInput
      type={type}
      id={id}
      name={name}
      value={value}
      className={`input ${classes ? classes : ''}`}
      placeholder={placeholder}
      disabled={disabled}
      onKeyUp={handleKeyUp}
      onChange={handleChange}
      onSubmit={submit}
    />

  );
};

const inputBaseStyle = css`
    align-items: center;
    border: ${rem(1)} solid ${colorPallete.light['80']};
    border-radius: ${rem(2)};
    box-shadow: none;
    display: inline-flex;
    font-size: ${rem(16)};
    height: ${rem(32)};
    justify-content: flex-start;
    line-height: $control-line-height;
    padding: ${rem(8)};
`;

const StyledDefaultInput = styled.input`
    ${inputBaseStyle}

    border-left-color: ${({ errors, name }) => {
    return errors && errors[name]
      ? 'red'
      : colorPallete.light['80'];
  }};
  border-left-width: ${({ errors, name }) =>
    errors && errors[name] ? rem(5) : rem(1)};

  &:hover,
  &:focus {
    border-color: ${({ errors, name }) => {
      return errors && errors[name]
        ? 'red'
        : colorPallete.light['87'];
    }};
    border-left-width: ${({ errors, name }) =>
      errors && errors[name] ? rem(5) : rem(1)};
  }
`;

export default DefaultInput;
