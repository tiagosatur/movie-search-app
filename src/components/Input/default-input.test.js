import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import DefaultInput from './DefaultInput';

describe('DefaultInput tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <DefaultInput />
    ).dive();
  });
  
  it('should render the DefaultInput element', () => {
    expect(wrapper.find('.input').exists()).toBe(true);
  });

  it('should call DefaultInput snapshot correctly', () => {
    const tree = renderer
        .create(wrapper)
        .toJSON();
    expect(tree).toMatchSnapshot();
    });
})