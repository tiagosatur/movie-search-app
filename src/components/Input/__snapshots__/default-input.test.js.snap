// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultInput tests should call DefaultInput snapshot correctly 1`] = `
<input
  className="input  sc-bdVaJa bgfZqT"
/>
`;
